doc/network.md

```diff
@@ -0,0 +1,325 @@
+| world  | Identifier | The world in which the workspace is present.                           |
+| data   | NBT Tag    | Arbitrary data assigned to the map template.                           |
+
+#### `plasmid:workspace/bounds`
+
+Sets the bounds of the workspace map. The client should render those bounds if non-null.
+A client can request bounds change, but can be rejected. If a request is accepted, the packet is sent back to the client.
+
+Direction `C<->S`
+
+| Fields | Type       | Description                                                                      |
+|:------:|:----------:|:---------------------------------------------------------------------------------|
+| id     | Identifier | The identifier of the affected workspace.                                        |
+| bounds | [Bounds]   | The new bounds of the workspace map, `[[0, 0, 0], [0, 0, 0]]` if bounds is null. |
+
+#### `plasmid:workspace/data`
+
+Sets the associated data on the workspace map.
+A client can request data change, but can be rejected. If a request is accepted, the packet is sent back to the client.
+
+Direction: `C<->S`
+
+| Fields | Type       | Description                               |
+|:------:|:----------:|:------------------------------------------|
+| id     | Identifier | The identifier of the affected workspace. |
+| data   | NBT Tag    | The NBT data.                             |
+
+#### `plasmid:workspace/region`
+
+Specifies a single region, the client should render a box, and a marker tag for it.
+If sent from a client, this is considered as an update request which can be rejected.
+
+Direction: `C<->S`
+
+| Fields    | Type       | Description                                                          |
+|:---------:|:----------:|:---------------------------------------------------------------------|
+| workspace | Identifier | The identifier of the workspace in which the region should be added. |
+| id        | [VarI32]   | The region runtime identifier.                                       |
+| region    | [Region]   | The region to add.                                                   |
+
+#### `plasmid:workspace/regions`
+
+Specifies regions to the client, for each region the client should render a box, and a marker tag.
+
+Direction: `C<-S`
+
+| Fields    | Type                              | Description                                                       |
+|:---------:|:---------------------------------:|:------------------------------------------------------------------|
+| workspace | Identifier                        | The identifier of the workspace containing the following regions. |
+| marker    | string                            | The marker of the regions to update.                              |
+| regions   | ([VarI32], [Bounds], NBT Tag)\[\] | The array of regions to update. The tuple is, in order, `(runtime id, bounds, data)`. |
```

Comment:
If a lot of regions with different markers were to be updated, would they be sent in separate packets?

Reply:
Yes, and even if they weren't, there's still the 1MB limit so I guess there's not much choice.